src/controller/java/src/chip/devicecontroller/ChipDeviceController.java

```diff
@@ -387,7 +387,29 @@ public void commissionDevice(long deviceId, @Nullable NetworkCredentials network
    */
   public void commissionDevice(
       long deviceId, @Nullable byte[] csrNonce, @Nullable NetworkCredentials networkCredentials) {
-    commissionDevice(deviceControllerPtr, deviceId, csrNonce, networkCredentials);
+    commissionDevice(deviceControllerPtr, deviceId, csrNonce, networkCredentials, null);
+  }
+
+  /**
+   * Initiates the automatic commissioning flow using the specified network credentials. It is
+   * expected that a secure session has already been established via {@link
+   * #establishPaseConnection(long, int, long)}.
+   *
+   * @param deviceId the ID of the node to be commissioned
+   * @param csrNonce a nonce to be used for the CSR request
+   * @param networkCredentials the credentials (Wi-Fi or Thread) to be provisioned
+   * @param icdRegistrationInfo the informations for ICD registration. For detailed information
+   *     {@link ICDRegistrationInfo}. If this value is null when commissioning an ICD device, {@link
+   *     CompletionListener.onICDRegistrationInfoRequired} is called to request the
+   *     ICDRegistrationInfo value.
+   */
+  public void commissionDevice(
```

Comment:
adding more and more parameters to a method generally does not scale well. Would it be possible to maybe have some builder to set parameters instead? So that I can write:

```
commissonDevice(
   CommissionParameters.Builder()
       .setCsrNonce(...)
       .setNetworkCredentials(...)
       .setIcdRegistrationInfo(...)
)
```

and then you can omit what you do not need.

Reply:
I agree. There are many modifications, but there are many similar methods in ChipDeviceController. Therefore, I thought it would be better to change them all together, so I changed them all. Please review.